### ISSUE TITLE
Navigation is not done when showing Alert in the Init() method

### ISSUE THREAD

**Issue by fcogutierrez** [rid00z/FreshMvvm#58]
Hello.

When I try to show an alert from the PageModel's **Init()** method (if there is any exception I want to show any UI alert), not only the navigation is not completed, but the navigation is "broken" for the rest to the application (so I can't do push to anywhere).

        public override void Init(object initData)
        {
            base.Init(initData);
            UserDialogs.Instance.Alert("Message", "Message content", "Close");            
        }

I have already tried without the UserDialogs plugin, and the error persists:

        ((MasterPage)Application.Current.MainPage).DisplayAlert("Message", "Message content", "Close");

Is this a bug? Any workaround?

Waiting for your replies. Thanks.

**Comment by fcogutierrez** [comment]
any solution?

**Comment by rid00z** [comment]
Init method is only used for setup, not for showing messages. 

It would be better to use the appearing events on the PageModel.

**Comment by fcogutierrez** [comment]
Ok, when I use the **ViewIsAppearing()** method it doesn't happen this.

@rid00z in my app, each time I navigate (doing push) to some pages, the PageModel needs to do some calls to server side (and will show some alert if there is some exception). Must I use **ViewIsAppearing** instead of **Init** method?

**Comment by fcogutierrez** [comment]
up

**Comment by rid00z** [comment]
Yes appearing events are better but you can also set of a background task in the init method.

**Comment by fcogutierrez** [comment]
Yes, the problem is where **in the Init method is there any exception** and, because of this exception, I want to show an alert. The navigation is broken and you cannot do any navigation.

For example, in the Init method I do something like...

        public override void Init(object initData)
        {
            base.Init(initData);

            try
            {                              
                CallToApi();
            }
            catch (Exception ex)
            {
                // Show a modal with some error message.
            }
        }

I have the feeling that I am doing something wrong...

Thanks.

**Comment by fcogutierrez** [comment]
any idea guys? :-)

**Comment by blackice2999** [comment]
Yeah, why you not call the api on Event? And create an activity indicator in init.

**Comment by rid00z** [comment]
Task.Factory.StartNew(() => DownloadDataFromInternet(), TaskCreationOptions.LongRunning);

**Comment by lloydkevin** [comment]
I'm had similar issues with Init().

Since this get's called _before_ the view model is set as the BindingContext on the page, then you can't perform any UI activities here.

I've had to resort to saving the parameter as a member variable, then doing any of my real work in the ViewIsAppearing()
